modules/deployment-model/src/main/java/org/mule/runtime/deployment/model/internal/application/MuleApplicationClassLoader.java

```diff
@@ -58,14 +60,36 @@ protected String findLibrary(String name) {
     return new String[] {toFile(classLoaderModel.getUrls()[0]).getPath()};
   }
 
+
+  /**
+   * Resolves the plugin classloader of the thread context class loader artifact and all it's ancestor
+   */
+  public static List<ClassLoader> resolveContextArtifactPluginClassLoaders() {
+    Set<ClassLoader> resolvedClassLoaders = new HashSet<>();
+    ClassLoader originalClassLoader = currentThread().getContextClassLoader();
+    ClassLoader tmpClassLoader = originalClassLoader;
+    resolvedClassLoaders.addAll(resolveContextArtifactPluginClassLoadersForCurrentClassLoader());
+    try {
```

Comment:
When you code, try to add empty spaces to separate different parts of the code to make it easier to read.